Chore: Creates new intermediate model int_server_hosting_type

sri-byte opened the pull request:
Impact: Creates a new intermediate model `int_server_hosting_type` so the logic for hosting type can be reused downstream in other dim and fact models.

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

ifoukarakis reviewed (changes requested):
`dim_server_info` on the branch contains info where `hosting_type` is null. This is not happening on the master branch, so a bug must have been introduced.

Also checked if there are differences in the hosting type between old and new approach. Only 3 servers seem to have a different hosting type.

catalintomai commented on transform/mattermost-analytics/models/intermediate/product/servers/int_server_hosting_type.sql:
```diff
@@ -11,11 +11,12 @@ with server_telemetry_summary as (
        coalesce(st.count_not_is_cloud_days, 0) as count_not_is_cloud_days
     from
-        server_telemetry_summary st
+        server_telemetry_summary st full outer join
```
nit: move "full outer join" statement on next row (in front of the "ref"), to be consistent across the codebase.

sri-byte replied:
Updated.

sri-byte commented:
> `dim_server_info` on the branch contains info where `hosting_type` is null. This is not happening on the master branch, so a bug must have been introduced.
> 
> Also checked if there are differences in the hosting type between old and new approach. Only 3 servers seem to have a different hosting type.

This is now updated, I missed a join to the user telemetry data which is now added.

sri-byte commented:
The results now align with master
